sktime/forecasting/naive.py

```diff
@@ -139,5 +143,40 @@ def _predict_last_window(self, fh, X=None, return_pred_int=False,
             fh_idx = fh.index_like(self.cutoff)
             return last_window[fh_idx]
 
-        elif self.strategy == "mean":
+        elif self.strategy == "mean" and self.sp == 1:
             return np.repeat(np.nanmean(last_window), len(fh))
+
+        elif self.strategy == "mean" and self.sp != 1:
+            last_window = last_window.astype(float)
+            last_window = np.pad(last_window,
+                                 (0, self.sp_ - len(last_window) % self.sp_),
+                                 'constant',
+                                 constant_values=(-1))
+
+            last_window = last_window.reshape(np.int(np.
+                                                     ceil(len(last_window) /
+                                                          self.sp_)),
+                                              self.sp_)
+
+            indices = np.where(last_window == -1)[-1]
+            last_window[-1][indices] =\
+                (last_window.sum(axis=0)[indices] +
+                 1) / (last_window.shape[-1] - 1)
```

Comment:
not sure what these lines of code are doing

Reply:
Here indices list is the occurences of all padded element's column indexes.
last_window[-1] is used to access the last row's elements with indices being thier coulmn indexes, and all padded elements belong to last row only .
here in this equation values are imputed to padded elements all at once.
the equation is desgined from solving for the 'x' mentioned in the last comment's example.
let sigma(ai) be sum of all previous values of this season, 'n' be no. of rows int last window matrix. what we need is:
{sigma(ai)+x}/n =sigma(ai)/(n-1); i.e mean of all the elements in the column of a season along with imputed padded value 'x' must return the mean of all the elemnts in the season column without padded value.